### DISCUSSION TITLE
ES9018

### DISCUSSION THREAD

**Discussion by alidaf** [pschatzmann/arduino-audio-tools#1371] [Q&A] [answered]
Have you tried an ES9018? They seem to be particularly painful to get working. I managed it on a RPi but I'm struggling to figure out whether it's even possible with an ESP32. They are normally configured via I2C but a lot of the cheap Chinese boards don't accommodate I2C access.

**Comment by pschatzmann** [comment]
I am not aware of any affordable module which would feature this chip.

**Reply by alidaf** [reply]
I used one of [these](https://www.aliexpress.com/item/1005003740290057.html?spm=a2g0o.order_list.order_list_main.42.15f8180213GQKT). There's various sellers offering the same thing but I went for the "deluxe" version, which looks like it has a higher spec clock as well as the shinier RCA sockets. It's probably ripped-off the Audiophonics board and there's no provenance to the actual DAC used but despite the instructions that implied it used the same config as a PCM5102, I only got it to work after using the RPi Audiophonics driver in Moode, which I think is the one [here](https://github.com/luoyi/Rpi-ES9018K2M-DAC).

**Comment by pschatzmann** [comment, marked as answer]
Interesting: Currently I don't have any time to add this to my [driver library](https://github.com/pschatzmann/arduino-audio-driver/tree/main/src/Driver)

[Maybe this library should help](https://github.com/possum64/BuffaloDAC)

**Reply by alidaf** [reply]
No worries. I was more after some insight so I'll try to see what the differences are between the drivers and DACs and learn something.

**Comment by pschatzmann** [comment]
I had a closer look at the description of this module and I am a little bit confused: I only see I2S pins.
I con't see any I2C pins or am I missing something ?

Did you try to output some sound via I2S e.g with an ESP32 ? 
Just make sure that the output is 32 bits and not 16...

**Reply by alidaf** [reply]
That's the point. Usually it is configured via I2C, but this board has no access. Anyway, I quite literally got it working about 30 minutes ago with no extra drivers using Squeezelite-ESP32. It just needed the 32-bit version as you also pointed out. I was using a faulty expansion board that also didn't help diagnosing the problem.

It has quite a nice sound to it. I need to sit with it for a while longer but my initial thoughts are that it seems quite a bit more analytical than the PCM5102. Possibly a little less bass but I'm listening on speakers that I designed (my first attempt using full range drivers) and they have something missing in the mid-bass anyway. The Op-Amp can also be swapped out for something better so I'm quite excited about this board.

**Reply by pschatzmann** [reply]
Cool: if you plan to use it with the AudioTools, you can do you processing still in 16bits and at the end just [change the format](https://github.com/pschatzmann/arduino-audio-tools/wiki/Converting-the-Data-Format) before you send it to I2S
